internal/core/id.go

```diff
@@ -2,11 +2,19 @@ package core
 
 import (
 	"fmt"
+
+	"github.com/google/uuid"
 )
 
-// ComponentID ... Represents a deterministic ID that's assigned
-// to all ETL components
-type ComponentID [4]byte
+// CPID ... Component Primary ID
+type CPID [4]byte
+
+// Represents a non-deterministic ID that's assigned to
+// every uniquely invoked component
+type ComponentID struct {
```

Comment:
I like this as it asserts within the name itself that it's unique (and is literally a combo of the component primary ID as well as UUID)
```suggestion
type ComponentUUID struct {
```

Reply:
[caac0d2](https://github.com/base-org/pessimism/pull/32/commits/caac0d2b0867fc8020fa713a6a3f197b78099704)